### PR TITLE
Fix 2 fakeintakes creation when not using load balancers

### PR DESCRIPTION
**Pull request by KevinFairise2**
What does this PR do?
---------------------

We had an issue where two fakeintake service where created when we want to create a fakeintake withtout load balancer. This PR solves this issue as well a refactoring a little bit the fakeintake service creation

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

File: components/datadog/apps/fakeintake/fargateFakeintakeService.go [hunk 1]
```diff
@@ -54,164 +55,25 @@ func NewECSFargateInstance(e aws.Environment, option ...fakeintakeparams.Option)
...
-			},
-		}
+	if params.LoadBalancerEnabled {
+		instance.Host, err = fargateServiceFakeIntakeWithLoadBalancer(e, namer, taskDef, opts...)
 	} else {
-		instance.Host, err = fargateServiceFakeintakeWithoutLoadBalancer(e, params.Name, params.ImageURL)
+		instance.Host, err = fargateServiceFakeintakeWithoutLoadBalancer(e, namer, taskDef)
 		if err != nil {
 			return nil, err
 		}
```
Comment (pducolin) [inline components/datadog/apps/fakeintake/fargateFakeintakeService.go:79]: **💬 suggestion**
You can move this err check out of the if

**Review by pducolin — APPROVED** [review]

File: components/datadog/apps/fakeintake/fargateFakeintakeService.go [hunk 2]
```diff
@@ -54,164 +55,25 @@ func NewECSFargateInstance(e aws.Environment, option ...fakeintakeparams.Option)
 	instance := &Instance{
 		Name: params.Name,
 	}
-	if err := e.Ctx.RegisterComponentResource("dd:fakeintake", namer.ResourceName("grp"), instance, opts...); err != nil {
+	if err := e.Ctx.RegisterComponentResource("dd:fakeintake", namer.ResourceName("grp"), instance, e.WithProviders(config.ProviderAWS, config.ProviderAWSX)); err != nil {
```
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply components/datadog/apps/fakeintake/fargateFakeintakeService.go]: Yes I deleted it in the commit just after the first [one](df509d92651c87e9617c7c79484b7cf37759c8ae).
This is indeed not needed and is exactly the same

File: components/datadog/apps/fakeintake/fargateFakeintakeService.go [hunk 3]
```diff
@@ -54,164 +55,25 @@ func NewECSFargateInstance(e aws.Environment, option ...fakeintakeparams.Option)
...
-			DefaultActions: clb.ListenerDefaultActionArray{
-				clb.ListenerDefaultActionArgs{
-					Type:           pulumi.String("forward"),
-					TargetGroupArn: alb.DefaultTargetGroup.Arn(),
-				},
-			},
-		}, opts...); err != nil {
-			return nil, err
-		}
+	taskDef, err := fargateLinuxTaskDefinition(e, namer, params.ImageURL)
```
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply components/datadog/apps/fakeintake/fargateFakeintakeService.go]: Good point, done!

File: components/datadog/apps/fakeintake/fargateFakeintakeService.go [hunk 4]
```diff
@@ -264,12 +126,9 @@ func fargateLinuxContainerDefinition(imageURL string) *ecs.TaskDefinitionContain
...
 // Hardcoded on sandbox
-func fargateServiceFakeintakeWithoutLoadBalancer(e aws.Environment, name, imageURL string) (ipAddress pulumi.StringOutput, err error) {
-	taskDef, err := fargateLinuxTaskDefinition(e, e.Namer.ResourceName(name, "taskdef"), imageURL)
-	if err != nil {
-		return pulumi.StringOutput{}, err
-	}
-	fargateService, err := ecsClient.FargateService(e, e.Namer.ResourceName(name, "srv"), pulumi.String(e.ECSFargateFakeintakeClusterArn()), taskDef.TaskDefinition.Arn())
+func fargateServiceFakeintakeWithoutLoadBalancer(e aws.Environment, namer namer.Namer, taskDef *ecs.FargateTaskDefinition) (ipAddress pulumi.StringOutput, err error) {
+
+	fargateService, err := ecsClient.FargateService(e, namer.ResourceName("srv"), pulumi.String(e.ECSFargateFakeintakeClusterArn()), taskDef.TaskDefinition.Arn())
```
[the remark this answers is not in the document]
Comment (KevinFairise2) [reply components/datadog/apps/fakeintake/fargateFakeintakeService.go]: Sounds good!

**Review by pducolin — APPROVED** [review]
Nice refactoring !

**Review by Pythyu — APPROVED** [review]
🚀

**Pull request merged**